numpy/core/src/umath/ufunc_object.c

```diff
@@ -565,7 +565,9 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
                     PyObject **out_typetup,
                     int *out_subok,
                     PyArrayObject **out_wheremask,
-                    PyObject **out_axes)
+                    PyObject **out_axes,
+                    PyObject **out_axis,
```

Comment:
A comment with the python type would be nice for this and the above line

Reply:
You mean something like
```
# list of tuples of specific axes for each argument
# axis to use for a shared, single core dimension
```
Am not sure it will help much to have that here.

Reply:
Yeah, something like that. Maybe even just `/* type: T */` and `/* type: List[Tuple[T]] */`, adopting PEP484 syntax (and considering @shoyer's remarks about non-int types